src/components/media-loader.js

```diff
@@ -80,35 +84,59 @@ AFRAME.registerComponent("media-loader", {
         this.blobURL = window.URL.createObjectURL(blob);
       }
 
-      if (contentType.startsWith("image/") || contentType.startsWith("video/") || contentType.startsWith("audio/")) {
+      if (
+        contentType.startsWith("image/") ||
+        contentType.startsWith("video/") ||
+        contentType.startsWith("audio/") ||
+        contentType === "application/pdf"
+      ) {
+        this.el.removeAttribute("gltf-model-plus");
         this.el.addEventListener(
           "image-loaded",
-          () => {
+          async () => {
             clearTimeout(this.showLoaderTimeout);
+            delete this.showLoaderTimeout;
+            if (contentType === "application/pdf") {
+              const maxIndex = await fetchMaxContentIndex(src, images.png);
+              this.el.setAttribute("media-pager", { index, maxIndex });
+            }
           },
           { once: true }
         );
-        this.el.setAttribute("image-plus", { src: this.blobURL || raw, contentType, token });
-        this.el.setAttribute("position-at-box-shape-border", { target: ".delete-button", dirs: ["forward", "back"] });
+        let imageSrc = contentType === "application/pdf" ? images.png : raw;
+        let imageContentType = contentType === "application/pdf" ? "image/png" : contentType;
+
+        if (contentType === "application/pdf") {
+          imageSrc = images.png;
+          imageContentType = "image/png";
+        } else {
+          this.el.removeAttribute("media-pager");
+        }
+
+        this.el.setAttribute("image-plus", { src: this.blobUrl || imageSrc, contentType: imageContentType, token });
+        this.el.setAttribute("position-at-box-shape-border", { dirs: ["forward", "back"] });
       } else if (
         contentType.includes("application/octet-stream") ||
         contentType.includes("x-zip-compressed") ||
         contentType.startsWith("model/gltf") ||
-        url.endsWith(".gltf") ||
-        url.endsWith(".glb")
+        src.endsWith(".gltf") ||
+        src.endsWith(".glb")
       ) {
+        this.el.removeAttribute("image-plus");
```

Comment:
Really not loving this sort of "make sure to do all the inverse opperations when media type switches" things. This is indicating to me that we still need to split this up more, but not 100% clear on the best way to do that yet.